packages/fast-usdc/src/type-guards.js

```diff
@@ -0,0 +1,21 @@
+import { M } from '@endo/patterns';
+
+/**
+ * @param {Brand} brand must be a 'nat' brand, not checked
+ * @param {NatValue} [min]
+ */
+export const makeNatAmountShape = (brand, min) =>
+  harden({ brand, value: min ? M.gte(min) : M.nat() });
+
+/** @param {Record<'PoolShares' | 'USDC', Brand<'nat'>>} brands */
+export const makeProposalShapes = ({ PoolShares, USDC }) =>
+  harden({
+    deposit: M.splitRecord(
+      { give: { ToPool: makeNatAmountShape(USDC, 1n) } },
+      { want: { Shares: makeNatAmountShape(PoolShares) } },
+    ),
+    withdraw: M.splitRecord({
+      give: { Shares: makeNatAmountShape(PoolShares, 1n) },
+      want: { FromPool: makeNatAmountShape(USDC, 1n) },
```

Comment:
On review, I see #10386 specifies names for these, which suggests:

```suggestion
    deposit: M.splitRecord(
      { give: { USDC: makeNatAmountShape(USDC, 1n) } },
      { want: { PoolShare: makeNatAmountShape(PoolShares) } },
    ),
    withdraw: M.splitRecord({
      give: { PoolShare: makeNatAmountShape(PoolShares, 1n) },
      want: { USDC: makeNatAmountShape(USDC, 1n) },
```

Reply:
Not a hard requirement. Is the style for keywords supposed to be their role in the transaction or what they are?

Reply:
> Not a hard requirement.

I went ahead and renamed them. Did some other refactoring too.

Keywords are supposed to be roles.